wordpress-mobile/WordPress-Android
# Date selector component

This PR extracts the date picker functionality into a separate `StatsDateSelector` class, which is then reused in the granular View all screens. 

The date selector is kept in sync with the currently selected date in the Stats and if the selection change, the data is reloaded on both screens.

![image](https://user-images.githubusercontent.com/1522856/54165914-03efa980-4463-11e9-9f26-c72310beefb2.png)

![image](https://user-images.githubusercontent.com/1522856/54165937-1538b600-4463-11e9-8b57-3f87d7b2a595.png)

To test:
1. Go to Stats and select a DMWY tab
2. Change currently selected date using the arrows in the date selector
3. Verify the date changes correctly and the label gets updated
4. Change the date by tapping on the graph
5. Verify correct date updates
6. Tap on the View all button under Posts and pages
7. Notice the date selector is present and correctly displays the current date
8. Change the selected date with the arrows
9. Notice the data gets refreshed according to the change
10. Go back to Stats
11. Notice the date is the same as it was in the View all screen and the data is refreshed
12. Switch to the Insights tab
13. Tap on the View more button under Followers
14. Notice the date selector is not visible
